src/imitation/scripts/train_rl.py

```diff
@@ -96,55 +96,66 @@ def train_rl(
-        # Normalize reward. Reward scale effectively changes the learning rate,
-        # so normalizing it makes training more stable. Note we do *not* normalize
-        # observations here; use the `NormalizeFeaturesExtractor` instead.
-        venv = VecNormalize(venv, norm_obs=False, **normalize_kwargs)
-        if reward_type == "RewardNet_normalized":
-            warnings.warn(
-                "Applying normalization to already normalized reward function. \
-                Consider setting normalize_reward as False",
-                RuntimeWarning,
+
+    try:
+        callback_objs = []
+        if reward_type is not None:
+            reward_fn = load_reward(
+                reward_type,
+                reward_path,
+                venv,
+                **load_reward_kwargs,
+            )
+            venv = RewardVecEnvWrapper(venv, reward_fn)
+            callback_objs.append(venv.make_log_callback())
+            logging.info(f"Wrapped env in reward {reward_type} from {reward_path}.")
+
+        if normalize_reward:
+            # Normalize reward. Reward scale effectively changes the learning rate,
+            # so normalizing it makes training more stable. Note we do *not* normalize
+            # observations here; use the `NormalizeFeaturesExtractor` instead.
+            venv = VecNormalize(venv, norm_obs=False, **normalize_kwargs)
+            if reward_type == "RewardNet_normalized":
+                warnings.warn(
+                    "Applying normalization to already normalized reward function. \
+                    Consider setting normalize_reward as False",
+                    RuntimeWarning,
+                )
+
+        if policy_save_interval > 0:
+            save_policy_callback = serialize.SavePolicyCallback(policy_dir)
+            save_policy_callback = callbacks.EveryNTimesteps(
+                policy_save_interval,
+                save_policy_callback,
             )
+            callback_objs.append(save_policy_callback)
+        callback = callbacks.CallbackList(callback_objs)
+
+        if agent_path is None:
+            rl_algo = rl.make_rl_algo(venv)
+        else:
+            rl_algo = rl.load_rl_algo_from_path(agent_path=agent_path, venv=venv)
+        rl_algo.set_logger(custom_logger)
+        rl_algo.learn(total_timesteps, callback=callback)
+
+        # Save final artifacts after training is complete.
+        if rollout_save_final:
+            save_path = osp.join(rollout_dir, "final.pkl")
+            sample_until = rollout.make_sample_until(
+                rollout_save_n_timesteps,
+                rollout_save_n_episodes,
+            )
+            types.save(save_path, rollout.rollout(rl_algo, venv, sample_until))
+        if policy_save_final:
+            output_dir = os.path.join(policy_dir, "final")
+            serialize.save_stable_model(output_dir, rl_algo)
+
+        # Final evaluation of expert policy.
+        result = train.eval_policy(rl_algo, venv)
```

Comment:
This is not necessary, you can return, and the finally still gets called anyways. (e.g. https://stackoverflow.com/questions/19805654/python-try-finally-block-returns)